api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -53,7 +53,12 @@ def render_error_payload(error, status = 422)
         end
 
         def spree_current_user
-          @spree_current_user ||= Spree.user_class.find_by(id: doorkeeper_token.resource_owner_id) if doorkeeper_token
+          return nil unless doorkeeper_token
+          return @spree_current_user if @spree_current_user
```

Comment:
Do we need that here? I think we should `doorkeeper_authorize!` every time when `doorkeeper_token` is present in case the token is expired.

Reply:
This early return is just to keep the preexisting memoization. Otherwise we would the DB would be hit on every `spree_current_user` call.

If `@spree_current_user` is set, it means that the token has been checked and is not expired.

Reply:
👌